### ISSUE TITLE
Certain valid regexes result in "internal fatal error" on compiling

### ISSUE THREAD

**Issue by deadpixi** [VirusTotal/yara#492]
The problem presents itself when regexes hit the RE_MAX_SPLIT_LIMIT, which can be done with fairly simple regexes. For example:

```
/((AA|BB|CC|DD|EE|FF|GG|HH|II)[A-Za-z0-9+]*x*a?){5,6}/
```

(Note that of course that regex could be written better, this is just for the purposes of triggering the bug.)

I would recommend if possible better error reporting, perhaps along the lines of "too many repeated alternations" or something, at least to better indicate the nature of the problem.

Thanks!

**Comment by pilate** [comment]
Another minimal example: 

```
rule example {
    strings:
        $a = /(1|2|3|4|5|6|7|8|9|0){8}/

    condition:
        $a
}
```
Changing {8} to {7} works fine, but anything higher errors out.

**Comment by plusvic** [comment]
This issue has been solved in version 3.6.0
